Source/Core/DolphinQt/Achievements/AchievementSettingsWidget.cpp

```diff
@@ -247,11 +247,7 @@ void AchievementSettingsWidget::ToggleRAIntegration()
     instance.Init();
   else
     instance.Shutdown();
-  if (Config::Get(Config::RA_HARDCORE_ENABLED))
-  {
-    emit Settings::Instance().EmulationStateChanged(Core::GetState(Core::System::GetInstance()));
-    emit Settings::Instance().HardcoreStateChanged();
-  }
+  ToggleHardcore();
```

Comment:
This is probably fine but I'm going to be a bit nitpicky here.

The main is sue is we're calling `SaveSettings()` twice.  Once at the beginning of this function and again in `ToggleHardcore()`.

I guess my suggestion would be to introduce a second function.  `UpdateHardcoreMode()`?  I'm terrible with names.  Though maybe the fact it is doing so much is why it is hard to name...

The changes would look like:

```
void AchievementSettingsWidget::ToggleRAIntegration()
{
  SaveSettings();
  auto& instance = AchievementManager::GetInstance();
  if (Config::Get(Config::RA_ENABLED))
    instance.Init();
  else
    instance.Shutdown();
  UpdateHardcoreMode();
}

void AchievementSettingsWidget::ToggleHardcore()
{
  SaveSettings();
  UpdateHardcoreMode()
}

void AchievementSettingsWidget::UpdateHardcoreMode()
{
  AchievementManager::GetInstance().SetHardcoreMode();
  if (Config::Get(Config::RA_HARDCORE_ENABLED))
  {
    if (Config::Get(Config::MAIN_EMULATION_SPEED) < 1.0f)
      Config::SetBaseOrCurrent(Config::MAIN_EMULATION_SPEED, 1.0f);
    Config::SetBaseOrCurrent(Config::FREE_LOOK_ENABLED, false);
    Config::SetBaseOrCurrent(Config::MAIN_ENABLE_CHEATS, false);
    Settings::Instance().SetDebugModeEnabled(false);
  }
  emit Settings::Instance().EmulationStateChanged(Core::GetState(Core::System::GetInstance()));
  emit Settings::Instance().HardcoreStateChanged();
}
```

What do you think?